pydb/database.py

```diff
@@ -0,0 +1,83 @@
+import abc
+
+from enum import Enum
+
+
+class DatabaseType(Enum):
+    """
+    not implemented: other databases
+    """
+    ORACLE = "oracle"
+    SQL_ALCHEMY_ORACLE = "sql_alchemy_oracle"
+
+
+class DBInterface(metaclass=abc.ABCMeta):
+    @classmethod
+    def __subclasshook__(cls, subclass):
+        return (hasattr(subclass, 'execute_query') and
+                callable(subclass.execute_query) and
+                hasattr(subclass, 'execute_update') and
+                callable(subclass.execute_update) and
+                hasattr(subclass, 'health_check') and
+                callable(subclass.health_check) and
+                hasattr(subclass, 'cleanup') and
+                callable(subclass.cleanup) and
+                hasattr(subclass, 'create_connection') and
+                callable(subclass.create_connection) and
+                hasattr(subclass, 'get_session') and
+                callable(subclass.get_session)
+                or NotImplemented)
```

Comment:
Is there a specific reason you're overriding this?

You already are handling this with the `abstractmethod` decorators. If a subclass fails to implement one of these it can't be instantiated.

Reply:
updated - removed